onnxruntime/core/framework/memory_info.cc

```diff
@@ -365,7 +365,7 @@ void MemoryProfiler::CreateEvents(const std::string& p_name,
 void MemoryProfiler::GenerateMemoryProfile() {
   // Write memory profile .json
   std::stringstream ss;
-  ss << "memory_profile_" << GetMemoryInfo().GetLocalRank() << "_" << profiler_id_ << ".json";
+  ss << "memory_profile_" << GetMemoryInfo().GetLocalRank() << "_" << Env::Default().GetSelfPid() << "_" << profiler_id_ << ".json";
```

Comment:
why this change?

Reply:
Oh this is because I find if we run this on two nodes, say 16 GPUs. profiler_id_ is always 0 for each process, local ranks are 0, 1, 2, 3, 4, 5, 6, 7, then, 0, 1, 2, 3, 4, 5, 6, 7. So the dumped data are overridden by the last write. I think we should use current process id as a unique key for this case.